ptvsd/wrapper.py

```diff
@@ -129,6 +130,47 @@ def using_format(self, fmt):
 str_handlers = pydevd_extutil.EXTENSION_MANAGER_INSTANCE.type_to_instance.setdefault(pydevd_extapi.StrPresentationProvider, [])  # noqa
 str_handlers.insert(0, SafeReprPresentationProvider._instance)
 
+DONT_TRACE_FILES = {
+    os.path.join('ptvsd', 'attach_server.py'),
+    os.path.join('ptvsd', 'daemon.py'),
+    os.path.join('ptvsd', 'debugger.py'),
+    os.path.join('ptvsd', 'futures.py'),
+    os.path.join('ptvsd', 'ipcjson.py'),
+    os.path.join('ptvsd', 'pathutils.py'),
+    os.path.join('ptvsd', 'pydevd_hooks.py'),
+    os.path.join('ptvsd', 'reraise.py'),
+    os.path.join('ptvsd', 'reraise2.py'),
+    os.path.join('ptvsd', 'reraise3.py'),
+    os.path.join('ptvsd', 'runner.py'),
+    os.path.join('ptvsd', 'safe_repr.py'),
+    os.path.join('ptvsd', 'socket.py'),
+    os.path.join('ptvsd', 'untangle.py'),
+    os.path.join('ptvsd', 'version.py'),
+    os.path.join('ptvsd', 'wrapper.py'),
+    os.path.join('ptvsd', '_main.py'),
+    os.path.join('ptvsd', '_version.py'),
+    os.path.join('ptvsd', '__init__.py'),
+    os.path.join('ptvsd', '__main__.py'),
+}
+
+
+def filter_ptvsd_files(file_path):
+    """
+    Returns true if the file should not be traced.
+    """
+    ptvsd_path_re = r"(ptvsd[\\\/].*\.py)"
+    matches = re.finditer(ptvsd_path_re, file_path)
+    for _, match in enumerate(matches):
+        for g in match.groups():
+            try:
+                return g in DONT_TRACE_FILES
+            except KeyError:
```

Comment:
You don't need the `except` anymore, singe `in` should never raise this for a missing key. 

Also, I think this entire loop can now become something like:
```py
return all((g in DONT_TRACE_FILES
            for m in matches
            for g in m.groups))
```

Reply:
Ah! yes i missed that. Will update.